R/utils.R

```diff
@@ -133,7 +133,11 @@ check_modules_datanames <- function(modules, datanames) {
         }
       )
     } else {
-      extra_datanames <- setdiff(modules$datanames, c("all", datanames))
+      extra_datanames <- unique(c(
+        setdiff(modules$datanames, c("all", datanames)),
+        setdiff(unlist(lapply(modules$transformers, function(x) x$datanames)), c("all", datanames))
+      ))
```

Comment:
Once this is finished https://github.com/insightsengineering/teal/issues/1298
This check can be extended